### ISSUE TITLE
Add support for images

### ISSUE THREAD

**Issue by rossant** [rossant/ipymd#37]
* Support Markdown image tags
* Support embedded images in the notebook
  * Extract images from the notebook
  * Save them to a given location, as specified by a customizable ImageManager class (for example, `images/image_003.png`)
  * Add a Markdown image tag

**Comment by rossant** [comment]
Further options for file naming:

* Increasing image index
* matplotlib's `title()`
* from the first comment in the cell
* from the plotting cell's metadata (`ipymd.image.filename` for example)

**Comment by Titan-C** [comment]
+1 on first comment in the cell.
title might contain latex of many characters that need escaping
The image index would make it hard to follow when reviewing the image files alone as there is no relation to the content.
If plot is redone would the system overwrite the image, or keep writing new ones?
